app/routes.py

```diff
@@ -240,16 +240,20 @@ def upgrade():
 @core.route('/environments')
 def environments():
     webhook = current_app.ldclient.variation('environments-webhook', current_user.get_ld_user(), False)
-
-    if webhook:
-        try:
-            ld = LaunchDarklyApi(os.environ.get('LD_API_KEY'))
-            project = ld.get_project(PROJECT_NAME)
-            project_pick = pickle.dumps(project)
-            current_app.redis_client.set(PROJECT_NAME, project_pick)
-            return jsonify({'response': 200})
-        except Exception as e:
-            current_app.logger.error(e)
-            return jsonify({'response': 400})
+    url = url_parse(request.url)
+    subdomain = url.host.split('.')[0]
+    if subdomain == 'admin':
+        if webhook:
+            try:
+                ld = LaunchDarklyApi(os.environ.get('LD_API_KEY'))
+                project = ld.get_project(PROJECT_NAME)
+                project_pick = pickle.dumps(project)
+                current_app.redis_client.set(PROJECT_NAME, project_pick)
+                return jsonify({'response': 200})
+            except Exception as e:
+                current_app.logger.error(e)
+                return jsonify({'response': 400})
+        else:
+            abort(404)
     else:
-        abort(404)
+        return "NO ACCESS"
```

Comment:
this should return the proper http status of unauthorized. I know I also need to fix jsonify to do it for the webhook itself.